website/docs/user_security.md

```diff
@@ -0,0 +1,294 @@
+            role = policy.value
+        if type(policy) == User.Status:
+            status = policy.value
+    if user_role <= role and user_status == status:
+        return True
+    else:
+        return False
+
+
+class CustomerSecurity(UserSecurity):
+    def __init__(self, cookie_name: str, secret: str, expiration_time: timedelta):
+        UserSecurity.__init__(self, cookie_name, JWTTokenHandler(secret, expiration_time))
+
+    def secure_entry(self, request, session_obj: SessionUser, policies: list):
+        data_dict = get_user(session_obj.id)
+        if data_dict['email'] == session_obj.email:
+            if not policies:
+                return response_status(HTTPStatus.OK)
+            elif has_access(data_dict, policies):
+                return response_status(HTTPStatus.OK)
+            else:
+                return response_status(HTTPStatus.UNAUTHORIZED)
+        else:
+            return response_status(HTTPStatus.FORBIDDEN)
+
+    def from_session_data(self, session_data: dict) -> SessionUser:
+        return SessionUser(session_data['id'], session_data['email'])
+
+    def generate_session_data(self, user) -> dict:
+        return {
+            'id': user['id'],
+            'email': user['email'],
+        }
+
+# CRUD INIT
+user_data_access = UserDataAccess()
+
+# SECURITY HANDLER REGISTER
+secret_key = 'super-secret'
+security_handler = CustomerSecurity('user_cookie', secret_key, timedelta(seconds=2))
+SecurityHandler.register(security_handler)
+
+
+# IMPLEMENTATION
+@RequireLogin(policies=[User.PolicyRoles.ADMIN, User.Status.ACTIVE])
+@handle_exceptions
+def admin_entry(request):
+    pass
+
+
+@RequireLogin(policies=[User.PolicyRoles.DELETE, User.Status.ACTIVE])
+@handle_exceptions
+def delete_entry(request):
+    pass
+
+
+@RequireLogin(policies=[User.PolicyRoles.CREATE, User.Status.ACTIVE])
+@handle_exceptions
+def create_entry(request):
+    pass
+
+
+@RequireLogin(policies=[User.PolicyRoles.UPDATE, User.Status.ACTIVE])
+@handle_exceptions
+def update_entry(request):
+    pass
+
+
+@RequireLogin(policies=[User.PolicyRoles.USER, User.Status.ACTIVE])
+@handle_exceptions
+def user_entry(request):
+    pass
+
+
+@RequireLogin(policies=[])
+@handle_exceptions
+def no_policy_entry(request):
+    pass
+# Function call with HttpRequest.COOKIES set as {'user_cookie': token}
+# where token is the JWT encoded token which will be decoded by UserSecurity
+# and the decoded object with data will be authenticated and authorized in secure_entry()
+response = user_entry(request)
+
+# Similarly, this process will happen with other function and the respective HTTP status code
+# will be returned. This is how UserSecurity is implemented in Core-lib
+
+# For authenticated user (Django example)
+@csrf_exempt
+@require_POST
+def api_login(request):
+    ...
+    body = request_body_dict(request)
+    email = body.get('email')
+    password = body.get('pass')
+    is_authenticated = auth_service.authnticate(email, password)

```

Comment:
rename it to 

`is_authenticated = core_lib.auth.authnticate(email, password)`